### PR TITLE
Make the endpoint id configurable

### PR DESCRIPTION
**Pull request by DaniFoldi**
Hi,

I'm glad this project exists, it simplified my deployment action a lot. I have a Portainer setup with multiple environments, and I had to deploy a stack to one that has an id > 1 (so not the default). I'm not that familiar with TS, so I'm pretty sure there are better ways to write this (hence the draft), however here is the code that I have now tested to be working.

### CODE REVIEW

**Comment by carlrygart** [comment]
Looks good! I actually considered adding this option when I created the project, but scoped it out since I didn't need it at the time. This is a great contribution! I will take a look later today so it works properly with my other setups (shouldn't be a problem) and then update all dependencies and bump the released version. Is that okay, or are you in a hurry? :)

**Comment by DaniFoldi** [comment]
Thanks! Please take your time testing it, I just thought that maybe the value should be optional, and default to 1 to maintain compatibility with existing workflows?

File: __tests__/deployStack.test.ts [hunk 1]
```diff
@@ -81,6 +81,7 @@ describe('deployStack', () => {
         username: 'username',
         password: 'password',
         swarmId: 's4ny2nh7qt8lluhvddeu9ulwl',
+        endpointId: '1',
```
Comment (carlrygart) [inline __tests__/deployStack.test.ts:84]: Let's create a proper test for this case.

File: action.yml [hunk 2]
```diff
@@ -18,12 +18,18 @@ inputs:
   swarm-id:
     required: false
     description: 'ID of the swarm. Only required if you deploy to a swarm.'
+  endpoint-id:
+    required: true
```
Comment (carlrygart) [inline action.yml:22]: Can we make this optional instead and then default to 1 to ensure backward compatible?

File: action.yml [hunk 3]
```diff
@@ -18,12 +18,18 @@ inputs:
...
+    description: 'ID of the Portainer node to deploy to'
   stack-name:
     required: true
     description: 'Name for the Portainer stack'
   stack-definition:
     required: true
     description: 'The path to the docker-compose stack stack definition file, eg. stack-definition.yml'
+  variables:
```
Comment (carlrygart) [inline action.yml:30]: Can we call this one `template-variables` instead?

File: src/deployStack.ts [hunk 4]
```diff
@@ -1,15 +1,18 @@
...
 import fs from 'fs'
+import Handlebars from 'handlebars'
 import * as core from '@actions/core'
 
 interface DeployStack {
   portainerHost: string
   username: string
   password: string
   swarmId?: string
+  endpointId: string
```
Comment (carlrygart) [inline src/deployStack.ts:12]: Let's make this one a number instead and then do the `parseInt` and set default to 1 inside `main.ts`.

File: src/deployStack.ts [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
   stackName: string
   stackDefinitionFile: string
+  variables?: string
```
Comment (carlrygart) [inline src/deployStack.ts:15]: Can we parse this one in `main.ts` instead?

File: src/deployStack.ts [hunk 6]
```diff
@@ -20,18 +23,26 @@ enum StackType {
...
-  const stackDefinition = fs.readFileSync(stackDefFilePath, 'utf8')
+  let stackDefinition = fs.readFileSync(stackDefFilePath, 'utf8')
   if (!stackDefinition) {
     throw new Error(`Could not find stack-definition file: ${stackDefFilePath}`)
   }
 
+  if (variables) {
+    core.info(`Applying variables: ${variables}`)
+    const variablesObject = JSON.parse(variables)
+    const template = Handlebars.compile(stackDefinition)
```
Comment (carlrygart) [inline src/deployStack.ts:42]: Very nice! :)

File: src/deployStack.ts [hunk 7]
```diff
@@ -20,18 +23,26 @@ enum StackType {
...
+    stackDefinition = template(variablesObject)
+  }
+
   if (!image) {
     core.info(`No new image provided, using existing stack definition`)
```
Comment (carlrygart) [inline src/deployStack.ts:47]: Change this to `No new image provided. Will use image in stack definition.`

**Review by carlrygart — CHANGES_REQUESTED** [review]
Really well done! Sorry I didn't have time to look at this earlier. I tried it out with my pipelines and it is not backwards compatible right now because of the required endpoint-id input. I suggest that we make this input optional and that we default to 1. I added some other comments as well. Do you want to fix them yourself or should I?

**Comment by carlrygart** [comment]
@DaniFoldi Do you want to fix these changes or do you want me to fix them for you? In that case, let me know if you think it's okay for me to push to your branch (which is your master branch) branch or if I should create a separate branch.

**Comment by DaniFoldi** [comment]
Hi, it's been a busy period for me, so if you have some free time, feel free to push any changes, if you get around to it

**Comment by DaniFoldi** [comment]
@carlrygart all the comments should have been implemented now, including a test case for endpoint ids

**Comment by Tim-Ganther** [comment]
Hi there, any updates regarding the merge? I am looking forward to the feature

**Review by carlrygart — APPROVED** [review]

**Comment by carlrygart** [comment]
@DaniFoldi Merged now :) Again, thanks a lot for the help! 👏 Will do some minor changes now a release a new version soon.

**Pull request merged**